.circleci/config.yml

```diff
@@ -56,35 +56,86 @@ workflows:
             - build
           <<: *job_filters
 
-      - architect/push-to-docker-legacy:
-          name: push-happa-to-aliyun-master
-          image: 'registry-intl.cn-shanghai.aliyuncs.com/giantswarm/happa'
-          username_envar: 'ALIYUN_USERNAME'
-          password_envar: 'ALIYUN_PASSWORD'
+      - architect/push-to-docker:
+          name: push-happa-to-quay
+          context: architect
+          image: "quay.io/giantswarm/happa"
+          username_envar: "QUAY_USERNAME"
+          password_envar: "QUAY_PASSWORD"
           requires:
             - build-container
+          # Needed to trigger job also on git tag.
           filters:
-            branches:
-              only: master
             tags:
-              only: /^v[0-9]+\.[0-9]+\.[0-9]+$/
-      - deploy-to-all:
+              only: /^v.*/
+
+      - architect/push-to-docker:
+          name: push-happa-to-aliyun
+          context: architect
+          image: "registry-intl.cn-shanghai.aliyuncs.com/giantswarm/happa"
+          username_envar: "ALIYUN_USERNAME"
+          password_envar: "ALIYUN_PASSWORD"
           requires:
             - build-container
-            - push-happa-to-aliyun-master
+          # Needed to trigger job only on git tag.
+          filters:
```

Comment:
Nit: The master branch filter can stay IMO. If someone needs to test it there they can remove it in a PR.